std/experimental/allocator/building_blocks/kernighan_ritchie.d

```diff
@@ -525,6 +530,15 @@ struct KRRegion(ParentAllocator = NullAllocator)
         return null;
     }
 
+    private void ensureCorrectSize()
+    {
+        import std.math : abs;
+
+        if (root && (abs(payload.length - root.size) < Node.sizeof)) {
```

Comment:
I'm slowly starting to understand :). This seems a bit more than necessary. The only situation when that happens is root is the same as payload and there's no next node:

```d
if (root == payload.ptr && root.size + Node.sizeof > payload.length)
{
    // Completely free, root comprehends the entire payload
    assert(!root.next || root.next == root);
    root.size = payload.length;
}
``` 

Is that correct? We don't need abs.